### DISCUSSION TITLE
Excessive use of toDecimal declaration while computing decimal columns with raw number

### DISCUSSION THREAD

**Discussion by calebeaires** [ClickHouse/ClickHouse#35364] [Q&A] [answered]
When multiplying a decimal column with a raw number described in the query, the execution is successful. But when division is used, execution returns an error.

```sql
SELECT MyDecimalNumberColumn * 2.2
-- Success
```

```sql
SELECT MyDecimalNumberColumn / 2.2
-- Illegal types Decimal(18, 2) and Float64 of arguments of function divide
```

My point is: **is there some Settings** to let Clickhouse understand that raw number described on querys is from a Decimal type. Sometimes we create big querys statements, use toDecimal64 (for example) for each time we use raw number is not so good.

**Comment by UnamedRus** [comment, marked as answer]
> is there some Settings to let Clickhouse understand that raw number described on querys is from a Decimal type.

No

```
SELECT MyDecimalNumberColumn / 2.2::Decimal64(2)
```

**Reply by calebeaires** [reply]
@UnamedRus, 

Thanks. We have this Business analytics app interely based in Clickhouse. I would like to know why to understand why it is this way. Could you tell me why Clickhouse has this rule!?

**Reply by UnamedRus** [reply]
> Could you tell me why Clickhouse has this rule!?

Because having Float64 is enough for 95% of use cases.
And this feature have low priority.  

https://github.com/ClickHouse/ClickHouse/issues/18841
